### PR TITLE
containerapp – fix error when running `az containerapp up` with no Dockerfile

### LINKED ISSUE #5893: containerapp – error running az containerapp up with no Dockerfile
- If the issue is to do with Azure CLI 2.0 in-particular, create an issue here at [Azure/azure-cli](https://github.com/Azure/azure-cli/issues)

### Related command

az containerapp up

### Extension name (the extension in question)

containerapp 0.3.22

### Description of issue (in as much detail as possible)

```
git clone https://github.com/Azure-Samples/msdocs-python-flask-webapp-quickstart
cd msdocs-python-flask-webapp-quickstart
az containerapp up -g myrg -n myapp --ingress external --source .
```

Prints error:

```
Running ACR build...
Failed to automatically generate a docker container from your source.
See the ACR logs above for more error information.
Please check the supported langauges for autogenerating docker containers 
```

-----

### PR DESCRIPTION
**Pull request by anthonychu**
Fixes #5893 

Looks like bug was introduced in #5749. File can't be deleted until `acr_task_run` is complete.

/cc @torosent @vinisoto 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az containerapp up

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
containerapp

**Review by StrawnSC — APPROVED** [review]

**Review by torosent — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@anthonychu Could you please resolve this CI issue?

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py [hunk 1]
```diff
@@ -100,6 +101,7 @@ def test_containerapp_identity_system(self, resource_group):
 
     @AllowLargeResponse(8192)
     @ResourceGroupPreparer(location="westeurope")
+    @live_only()
```
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:104]: Same as above

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 class ContainerappIdentityTests(ScenarioTest):
     @AllowLargeResponse(8192)
     @ResourceGroupPreparer(location="eastus2")
+    @live_only()
```
[the remark this answers is not in the document]
Comment (StrawnSC) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:22]: I think the test recording software is buggy in this case. We get "cannotoverwriteexistingcassette" exceptions in the CI, despite these tests passing when run locally

[the remark this answers is not in the document]
Comment (zhoxing-ms) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:22]: @StrawnSC Could you please show me the specific error message?

[the remark this answers is not in the document]
Comment (StrawnSC) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py]: I just removed the live_only, and as you can see we're getting the error in the CI again: `vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/containerapp/azext_containerapp/tests/latest/recordings/test_containerapp_identity_e2e.yaml') in your current record mode ('once').
No match for the request (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp000004?api-version=2022-01-31-preview>](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp000004?api-version=2022-01-31-preview%3E)) was found.`

[the remark this answers is not in the document]
Comment (StrawnSC) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py]: However, these tests pass when run locally:

Screen Shot 2023-03-03 at 10 26 52 AM

**Comment by zhoxing-ms** [comment]
> vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/containerapp/azext_containerapp/tests/latest/recordings/test_containerapp_identity_user.yaml') in your current record mode ('once').
No match for the request (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp-user1000004?api-version=2022-01-31-preview>](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp-user1000004?api-version=2022-01-31-preview%3E)) was found.
Found 1 similar requests with 1 different matcher(s) :

> 1 - (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp-user1000004?api-version=2023-01-31>).](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp-user1000004?api-version=2023-01-31%3E).)
Matchers succeeded : ['method', 'scheme', 'host', 'port', 'path']
Matchers failed :
_custom_request_query_matcher - assertion failure :
None

Please note that this problem is caused by the inconsistent api-version of `Microsoft.ManagedIdentity/userAssignedIdentities` in the CLI main repo. Related PR: https://github.com/Azure/azure-cli/pull/25418
This is because the [main repo](https://github.com/Azure/azure-cli) code in your local environment is too old. Please pull it from the remote dev branch, and then re-record these CI failed tests

**Comment by StrawnSC** [comment]
> > vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/containerapp/azext_containerapp/tests/latest/recordings/test_containerapp_identity_user.yaml') in your current record mode ('once').
> > No match for the request (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp-user1000004?api-version=2022-01-31-preview>](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp-user1000004?api-version=2022-01-31-preview%3E)) was found.
> > Found 1 similar requests with 1 different matcher(s) :
> 
> > 1 - (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp-user1000004?api-version=2023-01-31>).](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.ManagedIdentity/userAssignedIdentities/containerapp-user1000004?api-version=2023-01-31%3E).)
> > Matchers succeeded : ['method', 'scheme', 'host', 'port', 'path']
> > Matchers failed :
> > _custom_request_query_matcher - assertion failure :
> > None
> 
> Please note that this problem is caused by the inconsistent api-version of `Microsoft.ManagedIdentity/userAssignedIdentities` in the CLI main repo. Related PR: [Azure/azure-cli#25418](https://github.com/Azure/azure-cli/pull/25418) This is because the [main repo](https://github.com/Azure/azure-cli) code in your local environment is too old. Please pull it from the remote dev branch, and then re-record these CI failed tests

I rerecorded with the lastest version of the dev branch of the main CLI repo and it's still failing for those two tests in the CI

**Comment by zhoxing-ms** [comment]
> Successfully built containerapp
Installing collected packages: .....
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
azure-cli 2.46.0 requires azure-cli-core==2.46.0, but you have azure-cli-core 2.45.0 which is incompatible.

As we can see that the package `azure-cli-core` has dependency conflicts in the CI pipeline, this leads to the inconsistency of `azure-cli-core` version between the CLI pipeline and the local environment
Could you remove the redundant dependencies in `setup.py`? https://github.com/Azure/azure-cli-extensions/blob/970b414d7fb3635b4e57bd0576b762de699d9b89/src/containerapp/setup.py#L40

No explicit declaration is required for the dependency of `azure-cli-core`

**Comment by StrawnSC** [comment]
Looks like that fixed it. Thanks for the recommendation @zhoxing-ms! If we could get this merged soon, that'd be great -- we have some outstanding PRs that are waiting on this one

**Comment by anthonychu** [comment]
Thanks for your help @zhoxing-ms @StrawnSC! ❤️

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**